### ISSUE TITLE
Address forms don’t display select element for counties

### ISSUE THREAD

**Issue by MacLake** [mirumee/saleor#1335]
I have reported [this issue in google-i18n-address](https://github.com/mirumee/google-i18n-address/issues/24), but it turned out that it's actually a Saleor issue.

### What I'm trying to achieve
Customers  should be able to enter Irish addresses. Maybe other countries are also affected by this issue.

### Steps to reproduce the problem

1. Enter an Irish county in an address without knowing the exact strings expected.

### What I expected to happen
Display a select element for the county (or don't do validation checks on this field).

### What happened instead/how it failed
Counties are always shown as invalid unless spelled exactly like in the list provided by Google, which is hardly possible as these strings always start with „Co. “, see patrys’ comment in above mentioned issue.

**Comment by primal100** [comment]
My fix for this was to add a couple of lines at the beginning of update_base_fields to make the country area a choice field by adding the following lines at the beginning of the update_base_fields function.

``` 
def update_base_fields(form_class, i18n_rules):
  if i18n_rules.country_area_choices:
    form_class.base_fields['country_area'] = forms.ChoiceField(choices=i18n_rules.country_area_choices, required=False)```

This makes things better for a lot of countries. For example for the US, it now shows a list of states.

Does this look like a permanent solution? If so, I can make a pull request. I just had a brief look over the code so not sure if there is a better place to set this.

**Comment by patrys** [comment]
Because of how autocomplete works in browsers we are considering using an input with a `<datalist>` element instead of a select box here but the solution would be pretty similar in behavior.

**Comment by elwoodxblues** [comment]
Should be resolved by #1571
